File: determine_weirdness.py
def ltrim(s):
    return s.lstrip()

def rtrim(s):
    return s.rstrip()

def determine_weirdness(N):
    if N % 2 != 0:
        return "Weird"
    else:
        if 2 <= N <= 5:
            return "Not Weird"
        elif 6 <= N <= 20:
            return "Weird"
        else:
            return "Not Weird"

if __name__ == "__main__":
    N_temp = input("Enter a number: ")
    N = int(rtrim(ltrim(N_temp)))

    result = determine_weirdness(N)
    print(result)

File: DetermineWeirdness.java
import java.util.Scanner;

public class DetermineWeirdness {
    public static String ltrim(String str) {
        return str.replaceAll("^\\s+", "");
    }

    public static String rtrim(String str) {
        return str.replaceAll("\\s+$", "");
    }

    public static String determineWeirdness(int N) {
        if (N % 2 != 0)
            return "Weird";
        else {
            if (N >= 2 && N <= 5)
                return "Not Weird";
            else if (N >= 6 && N <= 20)
                return "Weird";
            else
                return "Not Weird";
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        String N_temp = scanner.nextLine();
        int N = Integer.parseInt(rtrim(ltrim(N_temp)));

        System.out.println(determineWeirdness(N));
    }
}

